lib/stable_email.py

```diff
@@ -49,9 +50,15 @@ def get_review_requests(dt_limit):
     repo = git.Repo(".")
     commits = repo.iter_commits("HEAD")
 
+    older_streak = 0
     for commit in commits:
         # Limit search
         if is_beyond_time_search(commit, dt_limit):
+            older_streak += 1
+        else:
+            older_streak = 0
+
+        if older_streak >= OLD_MSGS_STREAK:
```

Comment:
I hate this older_streak, and I'm trying to think of a better way to do it. The best idea that comes to mind is to create a function which provides the 'commits' generator for you and handles this logic. Something like the following:

```py
def get_commits_until(commits, dt_limit, old_msg_streak=10):
    older_streak = 0
    for commit in commits:
        if is_beyond_time_search(commit, dt_limit):
            yield commit
            older_streak += 1
        else:
            older_streak = 0
        if older_streak >= old_msg_streak:
            break
```

I'm not sure the logic there is exactly correct, and perhaps you want the function to handle the git repo stuff too. But this way, the complexity of the exit condition can be contained and documented. Then, your loop can just be `for commit in get_commits_until():` and you don't have to mix the loop exit logic with the commit handling stuff. It also removes the duplication since you had to copy/paste this algorithm in two places.

Reply:
maybe that yield should move to teh end of the loop rather than inside the if.

```
def get_commits_until(commits, dt_limit, old_msg_streak=10):
    older_streak = 0
    for commit in commits:
        if is_beyond_time_search(commit, dt_limit):
            older_streak += 1
        else:
            older_streak = 0
        if older_streak >= old_msg_streak:
            break
        yield commit
```